### PR TITLE
UI automation on Azure Pipeline

### PR DESCRIPTION
**Pull request by jfboismenu**
We can now run GUI automation tests using `MA.UI`. To enable UI testing on a repository, simply pass the `has_ui_tests` to `true` to the `build-pipeline.yml` include from your `azure-pipelines.yml` file.

Two new options for tests have been added to build-pipeline.yml, which replace `skip_tests`, `has_unit_tests` and `has_ui_tests`. Moving forward, we should use these two variables in `azure-pipelines.yml`. Note that `skip_tests` is kept for now and setting it to `true` still skips all tests, regardless of the value of the new `has_*_tests` parameters.

I've also refactored the ssh key install script and renamed the unit test template since they are not just about unit tests anymore.

### CODE REVIEW

File: build-pipeline.yml [hunk 1]
```diff
@@ -10,23 +10,42 @@
 
 parameters:
   # List of Python packages that need to be pip installed for testing.
-  extra_test_dependencies: []
+  - name: extra_test_dependencies
```
Comment (jfboismenu) [inline build-pipeline.yml:13]: Azure Pipelines now support fully type parameters for templates, so we'll document properly the supported parameters!

File: internal/run-tests.yml [hunk 2]
```diff
@@ -112,13 +87,46 @@ jobs:
...
+      python_version: 3.7
+      job_name: "macOS Python 3.7"
+      # pass through all parameters
+      ${{ insert }}: ${{ parameters }}
+
+  # Windows Python 3.7 build
+  - template: run-tests-with.yml
+    parameters:
+      image_name: 'windows-2019'
+      qt_wrapper: PySide2==5.14.1
```
Comment (pscadding) [inline internal/run-tests.yml:127]: Probably doesn't matter but aren't we targeting 5.12?

**Review by pscadding — APPROVED** [review]
Looks good to me!!

File: internal/run-tests.yml (hunk 2 above)
Comment (jfboismenu) [reply internal/run-tests.yml:127]: Yes. I pinned the version a while back to 5.14.1 due to a regression in Qt.

**Comment by jfboismenu** [comment]
UI automation tests can now be executed on Azure. To enable UI testing on a repository, set the "has_ui_tests" parameter to true in your azure-pipelines.yml file.

**Pull request merged**